pkg/http/http_test.go

```diff
@@ -165,7 +165,7 @@ func TestUploadRequest(t *testing.T) {
 		cookies       []*http.Cookie
 		expected      string
 	}{
-		{clientOptions: ClientOptions{}, method: "PUT", expected: "OK"},
+		{clientOptions: ClientOptions{MaxRetries: 3}, method: "PUT", expected: "OK"},
```

Comment:
Where does this become relevant in the test? Could we somehow validate the retry with a separate test?

Reply:
very valid point, just updated with a test